plugin.video.jupiterbroadcasting/default.py

```diff
@@ -0,0 +1,292 @@
+        feed = data['feed'] # High by default.
+        if quality == 1:
+            feed = data['feed-low']
+        elif quality == 2:
+            feed = data['feed-audio']
+        data['image'] = __get_show_image_path(data)
+        add_dir(item_name, feed, 1, data['image'], data)
+
+
+
+
+def translate_shows(__language__, untranslated_shows):
+    """
+    Takes the raw show input and translates the name and plot
+    """
+    translated_shows = {}
+    for show_name, data in untranslated_shows.items():
+        data['plot'] = __language__(data['plot'])
+        translated_shows[__language__(show_name)] = data
+
+    return translated_shows
+
+
+def index(name, url, page):
+    """
+    Presents a list of episodes within the given index page.
+    """
+
+    episodes_per_page = int(float(__settings__.getSetting('episodes_per_page')))
+
+    # Load feed and required info to figure out where
+    # to start and where to stop the pagination.
+    feedParser = FeedParser(name, url, episodes_per_page, page)
+    # Load/Parse the XML feed.
+    feedParser.parseXML()
+
+    # Wrap in a try/catch to protect from broken RSS feeds.
+    try:
+        for item in feedParser.getItems():
+
+            # Determine if item should be added based on
+            # page number and episodesPerPage
+            if feedParser.isItemBeforeCurrentPage():
+                # Skip this episode since it's before the page starts.
+                feedParser.nextItem()
+                continue
+            if feedParser.isPageEnd():
+                create_next_page_entry(__language__(30300), url, 1, page)
+                break
+
+            # Load up the initial episode information.
+            info = {}
+            info['title'] = feedParser.parseTitle(item)
+            info['tvshowtitle'] = name
+            info['count'] = feedParser.getCurrentFeedItem()
+            info['size'] = feedParser.parseVideoSize(item)
+            video = feedParser.parseVideo(item)
+            pubDate = feedParser.parsePubDate(item)
+            info['plotoutline'] = feedParser.parsePlotOutline(item)
+            info['plot'] = feedParser.parsePlot(item)
+            info['director'] = feedParser.parseAuthor(item)
+            thumbnail = feedParser.parseThumbnail(item)
+
+            # Get default thumb if needed except for archive/older episodes
+            if thumbnail == None and name != __language__(30025) and name != __language__(30300):
+                # Fall back to episode image
+                shows = translate_shows(__language__, jb_shows.get_all_shows())
+                thumbnail = __get_show_image_path(shows[name])
+
+            # Increment the show count.
+            feedParser.nextItem()
+
+            # Add the episode link.
+            add_link(info['title'], video, pubDate, thumbnail, info)
+    except Exception as e:
+        print 'Exception Parsing Feed! Please provide this log to https://github.com/RobLoach/plugin.video.jupiterbroadcasting'
```

Comment:
try to avoid `print` and prefer `xbmc.log`
If you can't avoid it change it at least to `print('Exception [...]')` which is python3 compatible